vernac/himsg.ml

```diff
@@ -114,6 +114,13 @@ let canonize_constr sigma c =
   in
   canonize_binders c
 
+let rec display_expr_eq c1 c2 =
+  let open Constrexpr in
+  match CAst.(c1.v, c2.v) with
+  | (CHole _ | CEvar _), _ | _, (CEvar _ | CHole _) -> true
```

Comment:
For the evar / evar case, you should also check the evar name, otherwise you might get extremely verbose output.

Reply:
You have a test case?

Reply:
I thought that frozen evars would have provided a test case but searching for a couple of minutes, I couldn't elaborate a test case. Nevertheless, I agree with @ppedrot that that would be "safer" to check the evar name.

Reply:
Without a test case to look at I still don't know what the issue is.

Reply:
I would think that @ppedrot had something like this is in mind, so that `?x1` and `?x2` are not considered equal:
```suggestion
  | CEvar (evk1,_) | CEvar (evk2,_) -> Evar.equal evk1 evk2
  | (CHole _ | CEvar _), _ | _, (CEvar _ | CHole _) -> true
```
If  `?x1` and `?x2` are not frozen, I'm finally doubting that they should be considered different. So, I'm rather leaning towards not changing the PR (@ppedrot??).

Reply:
I mean I want an example of an error message which is improved by that change.

Reply:
Improvement is subjective, but it's obvious that if you go with full printing throttle just because two evars are different may result in extremely verbose output. I am not sure why you're so adamant to have a minimal example when this can be demonstrated to happen from first principles.